iceoryx_examples/icedelivery/iox_publisher.cpp

```diff
@@ -54,7 +54,7 @@ int main()
         //  * Sample can be held until ready to publish.
         //  * Data is default constructed during loan
         publisher.loan()
-            .and_then([&](auto& sample) {
+            .and_then([&](auto& sample) -> void {
```

Comment:
If we agree on the trailing return type (instead of a template approach or ???), can you please add a FAQ entry about why this is necessary?